packages/itinerary-body/src/ItineraryBody/place-row.tsx

```diff
@@ -69,8 +71,48 @@ export default function PlaceRow({
     description: "Text describing the view-on-map button",
     id: "otpUi.ItineraryBody.viewOnMap"
   });
+
+  const formattedPlace = (
+    <PlaceName config={config} interline={interline} place={place} />
+  );
+  const time = isDestination ? leg.endTime : leg.startTime;
+  const a11yDetailsValues = {
+    mode: formattedModesByLeg
+      ? formattedModesByLeg[legIndex]
+      : getSummaryMode(leg, intl),
+    place: formattedPlace,
+    time: <FormattedTime value={time} />
+  };
+
   return (
     <S.PlaceRowWrapper key={legIndex || "destination-place"}>
+      <S.InvisibleAdditionalDetails>
+        {!isDestination ? (
+          <FormattedMessage
+            description="Invisible description of leg for screen readers"
+            id={
+              leg.routeShortName
+                ? "otpUi.TransitLegBody.accessibilityTransitLegDesc"
+                : "otpUi.TransitLegBody.accessibilityLegDesc"
```

Comment:
We could even just say
```
6:44 pm, from <location>, walk 300 feet to <next location>
```

Reply:
> We could even just say
> 
> ```
> 6:44 pm, from <location>, walk 300 feet to <next location>
> ```

I like this! I kept the `formattedModesByLeg` because without the visual `modeIcon` in the `TransitLeg`s, the mode may not be entirely clear, especially for infrequent riders. 

The `AccessLeg`s are really simple, I removed the invisible labels entirely from those legs. For the `TransitLeg`s it got a bit trickier, and I'm interested in seeing what you think. I added headsigns and destinations to the descriptions of each leg, and I'm worried the format of `routeName to headsign to destination` gets a little confusing.

Reply:
I like this a lot. This is an easy fix for the non-transit legs. The transit leg descriptions are a bit more complicated, I'm curious to get your thoughts!

I kept the `formattedModesByLeg` in, since transit legs don't otherwise show the route mode.

Reply:
@amy-corson-ibigroup Each place still needs to be read out, as in
```
Depart at 6:44 PM from KGW Studio on the Sq, Portland, OR, USA, Walk 269 feet...
```

Reply:
Regarding `formattedModesByLeg`, I still maintain we probably don't need that. I think this can be done using the `LegIcon` component. We can pass props such as `aria-label` that will be passed to the inner `<svg>` that these render into.